### PR TITLE
feat(API-1904): Add JS code snippets for GT "How to collect product variations"

### PR DESCRIPTION
**Pull request by samokiss**

### CODE REVIEW

File: content/tutorials/guides/how-to-collect-product-variations.md [hunk 1]
```diff
@@ -224,30 +294,69 @@ Query the API.
...
+    $indexedFamilyVariants[$familyVariant['code']] = $familyVariant;
+}
+
+// Save family variants into storage
+saveFamilyVariants($indexedFamilyVariants);
+
+```
+```javascript [activate:NodeJS]
+
+const maxItems = 100;
```
Comment (tomglvng) [inline content/tutorials/guides/how-to-collect-product-variations.md:333]: In PHP snippet, it is  `$maxProductsPerPage`
Harmonize it plz

File: content/tutorials/guides/how-to-collect-product-variations.md [hunk 2]
```diff
@@ -297,58 +428,102 @@ In the guided tutorial <a href="/tutorials/how-to-get-families-and-attributes.ht
...
-            )
-        );
-        $data = json_decode($response->getBody()->getContents(), true);
-        $productModels[] = $data['_embedded']['items'];
-    }
+$client = buildApiClient();
+
+$maxProductsPerPage = 100;
+$maxFamiliesPerQuery = 10;
+$scope = 'ecommerce';
```
Comment (tomglvng) [inline content/tutorials/guides/how-to-collect-product-variations.md:435]: Get from the storage (in PHP snippet also)

**Review by tomglvng — APPROVED** [review]
GTM after minor fixes

File: content/tutorials/guides/how-to-collect-product-variations.md [hunk 3]
```diff
@@ -130,84 +148,136 @@ In the guided tutorial <a href="/tutorials/how-to-get-families-and-attributes.ht
...
+$productModels = array_merge(...$productModels);
+
+// Save product models into storage
+saveProductModels($productModels);
+```
+```javascript [activate:NodeJS]
+
+const maxProductsPerPage = 100;
+const maxFamiliesPerQuery = 10;
+const scope = 'ecommerce';
```
[the remark this answers is not in the document]
Comment (samokiss) [reply content/tutorials/guides/how-to-collect-product-variations.md:196]: I dont think we have stored this value

[the remark this answers is not in the document]
Comment (samokiss) [reply content/tutorials/guides/how-to-collect-product-variations.md:196]: we decided to fix this into refacto ticket

File: content/tutorials/guides/how-to-collect-product-variations.md (hunk 2 above)
Comment (samokiss) [reply content/tutorials/guides/how-to-collect-product-variations.md:435]: we decided to fix this into refacto ticket

**Pull request merged**